src/server/db_slice.cc

```diff
@@ -351,32 +351,34 @@ DbSlice::AddOrFindResult& DbSlice::AddOrFindResult::operator=(ItAndUpdater&& o)
   return *this;
 }
 
-DbSlice::ItAndUpdater DbSlice::FindMutable(const Context& cntx, string_view key,
-                                           bool load_external) {
-  auto [it, exp_it] = FindInternal(cntx, key, FindInternalMode::kUpdateMutableStats, load_external);
-
-  if (IsValid(it)) {
-    PreUpdate(cntx.db_index, it);
-    return {it, exp_it,
-            AutoUpdater({.action = AutoUpdater::DestructorAction::kRun,
-                         .db_slice = this,
-                         .db_ind = cntx.db_index,
-                         .it = it,
-                         .key = key})};
-  } else {
-    return {it, exp_it, {}};
-  }
+DbSlice::ItAndUpdater DbSlice::FindAndFetchMutable(const Context& cntx, string_view key) {
+  return std::move(
+      FindMutableInternal(cntx, key, std::nullopt, LoadExternalMode::kLoadExternal).value());
+}
+
+DbSlice::ItAndUpdater DbSlice::FindMutable(const Context& cntx, string_view key) {
+  return std::move(
+      FindMutableInternal(cntx, key, std::nullopt, LoadExternalMode::kDontLoadExternal).value());
 }
 
 OpResult<DbSlice::ItAndUpdater> DbSlice::FindMutable(const Context& cntx, string_view key,
-                                                     unsigned req_obj_type, bool load_external) {
-  // Don't use FindMutable() so that we don't call PreUpdate()
-  auto [it, exp_it] = FindInternal(cntx, key, FindInternalMode::kUpdateMutableStats, load_external);
+                                                     unsigned req_obj_type) {
+  return FindMutableInternal(cntx, key, req_obj_type, LoadExternalMode::kDontLoadExternal);
+}
 
+OpResult<DbSlice::ItAndUpdater> DbSlice::FindAndFetchMutable(const Context& cntx, string_view key,
+                                                             unsigned req_obj_type) {
+  return FindMutableInternal(cntx, key, req_obj_type, LoadExternalMode::kLoadExternal);
+}
+
+OpResult<DbSlice::ItAndUpdater> DbSlice::FindMutableInternal(const Context& cntx, string_view key,
+                                                             std::optional<unsigned> req_obj_type,
+                                                             LoadExternalMode load_mode) {
+  auto [it, exp_it] = FindInternal(cntx, key, UpdateStatsMode::kUpdateMutableStats, load_mode);
```

Comment:
another reason to pass `std::optional<unsigned> req_obj_type` to FindInternal, is that this function becomes redundant.